internal/validation/mock_trace_validation.go

```diff
@@ -63,22 +65,30 @@ func validateTimeStamps(span *cloudtrace.Span) error {
 	return nil
 }
 
-func AddSpans(uploadedSpans map[string]*cloudtrace.Span, lock *sync.Mutex, spans ...*cloudtrace.Span) error {
-	br := &errdetails.ErrorInfo{}
+func AddSpans(uploadedSpans map[string]*cloudtrace.Span, lock *sync.Mutex,
+	delay time.Duration, ctx context.Context, spans ...*cloudtrace.Span) error {
 
+	br := &errdetails.ErrorInfo{}
 	lock.Lock()
```

Comment:
Don't hold the lock and make a blocking call. This will severely degrade performance if we have concurrent requests. Wait for <-time.After(delay) to return before you take the lock.

Reply:
Yep made the change, sorry for missing this